README.md

```diff
@@ -17,35 +18,53 @@ Your charm directory should have the following overall structure:
 |   +-- charm.py
 +-- hooks/
     +-- install -> ../src/charm.py
+    +-- start -> ../src/charm.py  # for k8s charms per below
 ```
 
-The `mod/` directory will contain the operator framework dependency as a git
+The `mod/` directory should contain the operator framework dependency as a git
 submodule:
 
 ```
 git submodule add https://github.com/canonical/operator mod/operator
 ```
 
-Other dependencies included as git submodules can be added there as well.
-
-The `lib/` directory will then contain symlinks to subdirectories of your
-submodule dependencies to enable them to be imported into the charm:
+Then symlink from the git submodule for the operator framework into the `lib/`
+directory of your charm so it can be imported at run time:
 
 ```
 ln -s ../mod/operator/ops lib/ops
 ```
 
+Other dependencies included as git submodules can be added in the `mod/`
+directory and symlinked into `lib/` as well.
+
+You can sync subsequent changes from the framework and other submodule
+dependencies by running:
+
+```
+git submodule update
+```
+
+Those cloning and checking out the source for your charm for the first time
+will need to run:
+
+```
+git submodule update --init
+```
+
 Your `src/charm.py` is the entry point for your charm logic. It should be set
-to executable and use Python 3.6 or greater. At a minimum, it needs to define a
-subclass of `CharmBase` and pass that into the framework's `main` function:
+to executable and use Python 3.6 or greater (as such, operator charms can only
```

Comment:
I wouldn't tie it to just Ubuntu by wording it like this.

How about:

> For Ubuntu it means that the 18.04 release or higher is required. Juju supports other OSes and if Python3.6 is available there out of the box or in a customized image then it will also be possible to use this framework.

Reply:
How about just keeping the way it is right now?  We require Python 3.6+, and that's it. Everyone can figure their own environment.